Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow.lean

```diff
@@ -0,0 +1,339 @@
+  rw [sqrt_eq_rpowₙ, rpowₙ_rpowₙ, one_div_mul_eq_div 2 x]
+
+lemma rpowₙ_sqrt_two {a : A} (ha : 0 ≤ a := by cfc_tac) : rpowₙ (sqrt a) 2 = a := by
+  simp only [rpowₙ_sqrt, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, div_self]
+  rw [rpowₙ_one]
+
+lemma sqrt_mul_sqrt_self {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt a * sqrt a = a := by
+  rw [← rpowₙ_two, rpowₙ_sqrt_two]
+
+@[simp]
+lemma sqrt_rpowₙ {a : A} {x : ℝ≥0} : sqrt (rpowₙ a x) = rpowₙ a (x / 2) := by
+  simp [sqrt_eq_rpowₙ, div_eq_mul_inv]
+
+lemma sqrt_rpowₙ_two {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt (rpowₙ a 2) = a := by
+  simp only [sqrt_rpowₙ, ne_eq, OfNat.ofNat_ne_zero, not_false_eq_true, div_self]
+  rw [rpowₙ_one]
+
+lemma sqrt_mul_self {a : A} (ha : 0 ≤ a := by cfc_tac) : sqrt (a * a) = a := by
+  rw [← rpowₙ_two, sqrt_rpowₙ_two]
+
+end sqrt
+
+end NonUnital
+
+section Unital
+
+variable {A : Type*} [PartialOrder A] [NormedRing A] [StarRing A] [StarOrderedRing A]
+  [TopologicalRing A] [NormedAlgebra ℂ A] [CompleteSpace A]
+  [ContinuousFunctionalCalculus ℝ≥0 (fun (a : A) => 0 ≤ a)]
+
+/- ## `rpow` -/
+
+/-- Real powers of operators, based on the unital continuous functional calculus. -/
+noncomputable def rpow (a : A) (y : ℝ) : A := cfc (fun x : ℝ≥0 => x ^ y) a
+
+/-- Enable `a ^ y` notation for `CFC.rpow`. This is a low-priority instance to make sure it does
+not take priority over other instances when they are available. -/
+noncomputable instance (priority := 100) : Pow A ℝ where
+  pow a y := rpow a y
+
+@[simp]
+lemma rpow_nonneg {a : A} {y : ℝ} : 0 ≤ a ^ y := cfc_predicate _ a
+
+@[simp]
+lemma rpow_eq_pow {a : A} {y : ℝ} : rpow a y = a ^ y := rfl
+
+lemma rpow_def {a : A} {y : ℝ} : a ^ y = cfc (fun x : ℝ≥0 => x ^ y) a := rfl
+
+lemma rpow_natCast {a : A} {n : ℕ} (ha : 0 ≤ a := by cfc_tac) : a ^ (n : ℝ) = a ^ n := by
+  have h₁ : (fun x : ℝ≥0 => x ^ (n : ℝ)) = fun (x : ℝ≥0) => x ^ n := by ext; simp
+  simp_rw [rpow_def, h₁]
+  change cfc (fun x : ℝ≥0 => (id x) ^ n) a = a ^ n
+  rw [cfc_pow (id : ℝ≥0 → ℝ≥0) n a, cfc_id ℝ≥0 a]
+
+@[simp]
+lemma rpow_algebraMap {x : ℝ≥0} {y : ℝ} :
+    (algebraMap ℝ≥0 A x) ^ y = algebraMap ℝ≥0 A (x ^ y) := by
+  simp only [rpow_def]
+  rw [cfc_algebraMap x (fun z : ℝ≥0 => z ^ y)]
+
+lemma rpow_add_of_zero_not_mem_spectrum {a : A} {x y : ℝ} (ha : 0 ∉ spectrum ℝ≥0 a) :
+    a ^ (x + y) = a ^ x * a ^ y := by
+  simp only [rpow_def, NNReal.rpow_eq_pow]
+  have h : ∀ r, ContinuousOn (fun z : ℝ≥0 => z ^ (r : ℝ)) (spectrum ℝ≥0 a) := by
+    intro r z hz
+    exact ContinuousAt.continuousWithinAt <| NNReal.continuousAt_rpow_const <| Or.inl <| by aesop
+  rw [← cfc_mul _ _ a (h x) (h y)]
+  refine cfc_congr ?_
+  intro z hz
+  have : z ≠ 0 := by aesop
+  simp [NNReal.rpow_add this _ _]
+
+-- TODO: relate to a strict positivity condition
+lemma rpow_rpow_of_spectrum_pos [UniqueContinuousFunctionalCalculus ℝ≥0 A]
+    {a : A} {x y : ℝ} (ha₁ : ∀ z ∈ spectrum ℝ≥0 a, 0 < z) (hx : x ≠ 0) (ha₂ : 0 ≤ a := by cfc_tac) :
```

Comment:
Since you already have the `0 ≤ a` assumption, you can get away with:
```suggestion
    {a : A} {x y : ℝ} (ha₁ : 0 ∉ spectrum ℝ≥0 a) (hx : x ≠ 0) (ha₂ : 0 ≤ a := by cfc_tac) :
```